src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -98,26 +98,125 @@ uct_cuda_copy_get_mem_type(uct_md_h md, void *address, size_t length)
     return mem_info.type;
 }
 
-static UCS_F_ALWAYS_INLINE ucs_status_t uct_cuda_copy_ctx_rsc_get(
-        uct_cuda_copy_iface_t *iface, uct_cuda_copy_ctx_rsc_t **ctx_rsc_p)
+static ucs_status_t
+uct_cuda_primary_ctx_push_first_active(CUdevice *cuda_device_p,
+                                       CUcontext *cuda_ctx_p)
+{
+    int num_devices, device_num;
+    ucs_status_t status;
+    CUdevice device;
+    CUcontext ctx;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGetCount(&num_devices));
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    for (device_num = 0; device_num < num_devices; ++device_num) {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&device, device_num));
+        if (status != UCS_OK) {
+            return status;
+        }
+
+        status = uct_cuda_primary_ctx_retain(device, 0, &ctx);
+        if (status == UCS_OK) {
+            break;
+        } else if (status == UCS_ERR_NO_DEVICE) {
+            continue;
+        } else {
+            return status;
+        }
```

Comment:
maybe do smth like
````c
if (status == UCS_OK) {
    /* Found active context */
    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxPushCurrent(ctx));
    if (status != UCS_OK) {
        UCT_CUDADRV_FUNC_LOG_WARN(cuDevicePrimaryCtxRelease(device));
        return status;
    }

    *cuda_device_p = device;
    *cuda_ctx_p    = ctx;
    return UCS_OK;
} else if (status != UCS_ERR_NO_DEVICE) {
    return status;
}
````

then no need to check for (device_num == num_devices) after the loop

Reply:
Applied suggestion.